spec/features/user_adds_review_spec.rb

```diff
@@ -0,0 +1,84 @@
+require "rails_helper"
+
+feature "users can add a new review for a piece of music" do
+  let(:piece) do
+    FactoryGirl.create(:piece)
+  end
+
+  context "not logged in user" do
+    scenario "cannot write review" do
+      visit piece_path(piece)
+      click_button 'Add Review'
+
+      expect(page).to have_content "Please sign in to add a review"
+    end
+  end
+
+  context "logged in user" do
+    let(:user) do
+      FactoryGirl.create(:user)
+    end
+
+    before do
+      visit root_path
+      click_link 'Sign In'
+      fill_in 'Email', with: user.email
+      fill_in 'Password', with: user.password
+      click_button 'Sign In'
+    end
```

Comment:
Good way to DRY this up since you need signed in users. However, you should create a `sign_in_as` helper method for you tests. Here's an [example] (https://github.com/davidrf/adventure_planner/blob/master/spec/support/helpers/authentication_helper.rb). I would begin the method with `visit root_path`. You will need to also need to require the file where you define this module and include it in Rspec. Take a look at lines 25 and 54 of [this `rails_helper.rb` file](https://github.com/davidrf/adventure_planner/blob/master/spec/rails_helper.rb) for guidance.